stingray/multitaper.py

```diff
@@ -0,0 +1,503 @@
+
+            err -= d_k
+            if np.max(np.mean(err ** 2, axis=0)) < 1e-10:
+                break
+
+            # update the iterative estimate with this d_k
+            psd_iter = self.psd_from_freq_response(freq_response, d_k)
+            err = d_k
+        if ite == max_iter - 1:
+            simon('Iterative multi-taper PSD computation did not converge.')
+
+        return psd_iter, d_k
+
+    def _normalize_multitaper(self, unnorm_power, tseg):
+        """
+        Normalize the real part of the mulitaper spectrum estimate to Leahy, 
+        absolute rms^2, fractional rms^2 normalization, or not at all.
+
+        Parameters
+        ----------
+        unnorm_power: numpy.ndarray
+            The unnormalized spectrum estimate.
+
+        tseg: int
+            The length of the Fourier segment, in seconds.
+
+        Returns
+        -------
+        power: numpy.nd.array
+            The normalized spectrum estimate (real part of the spectrum).
+            For 'none' normalization, imaginary part is returned as well.
+        """
+
+        if self.err_dist == 'poisson':
+            return normalize_crossspectrum(
+                unnorm_power, tseg, self.n, self.nphots, self.nphots, self.norm,
+                self.power_type)
+
+        return normalize_crossspectrum_gauss(
+            unnorm_power, np.sqrt(self.meancounts * self.meancounts),
+            np.sqrt(self.var * self.var),
+            dt=self.dt,
+            N=self.n,
+            norm=self.norm,
+            power_type=self.power_type)
+
+    def jackknifed_sdf_variance(self, freq_response, eigvals, adaptive):
+        r"""
+        Returns the variance of the log-sdf estimated through jack-knifing
+        a group of independent sdf estimates.
+        Parameters
+        ----------
+        yk : ndarray (K, L)
+        The K DFTs of the tapered sequences
+        eigvals : ndarray (K,)
+        The eigenvalues corresponding to the K DPSS tapers
+        sides : str, optional
+        Compute the jackknife pseudovalues over as one-sided or
+        two-sided spectra
+        adpative : bool, optional
+        Compute the adaptive weighting for each jackknife pseudovalue
+        Returns
+        -------
+        var : The estimate for log-sdf variance
+        Notes
+        -----
+        The jackknifed mean estimate is distributed about the true mean as
+        a Student's t-distribution with (K-1) degrees of freedom, and
+        standard error equal to sqrt(var). However, Thompson and Chave [1]
+        point out that this variance better describes the sample mean.
+        [1] Thomson D J, Chave A D (1991) Advances in Spectrum Analysis and Array
+        Processing (Prentice-Hall, Englewood Cliffs, NJ), 1, pp 58-113.
+        """
+
+        K = len(eigvals)
+
+        all_indices = set(range(K))
+        jk_sdk = []
+
+        for i in range(K):
+            selected_indices = list(all_indices.difference([i]))
```

Comment:
`selected_indices` here is in an empty list when the parameter NW <= 1.242. This results in an empty list being sent to the function call on lines 401 or 406, causing a divide by zero Runtime Error

Reply:
Can't you just force NW to be > 1.242?

Reply:
The PSD is calculated just fine with values below 1.242, that is, with just 1 Taper, but the jack-knife part produces nan value. But does jack-knife even make sense with 1 DPSS taper? As it is concerned with selectively calculating the PSD's of a subset of the tapered data, then combining these PSDs from subsets, but with one taper, there is no subset.